tests/acceptance/features/webUIUpload/upload.feature

```diff
@@ -20,17 +20,36 @@ Feature: File Upload
 
   @smokeTest @ocisSmokeTest
   Scenario: simple upload of a folder that does not exist before
-    Given a folder "CUSTOM" has been created with the following files in separate sub-folders in the server
-      | lorem.txt     |
-      | new-lorem.txt |
+    Given a folder "CUSTOM" has been created with the following files in separate sub-folders in the middleware
+      | subFolder | file          |
+      |           | lorem.txt     |
+      | sub1      | lorem.txt     |
+      | sub1      | new-lorem.txt |
+      | sub2/sub3 | new-lorem.txt |
     When the user uploads folder "CUSTOM" using the webUI
     Then no message should be displayed on the webUI
     And folder "CUSTOM" should be listed on the webUI
     And as "Alice" folder "CUSTOM" should exist in the server
     And as "Alice" file "CUSTOM/lorem.txt" should exist in the server
-    And as "Alice" file "CUSTOM/new-lorem.txt" should exist in the server
+    And as "Alice" file "CUSTOM/sub1/lorem.txt" should exist in the server
+    And as "Alice" file "CUSTOM/sub1/new-lorem.txt" should exist in the server
+    And as "Alice" file "CUSTOM/sub2/sub3/new-lorem.txt" should exist in the server
     And as "Alice" the content of "CUSTOM/lorem.txt" in the server should be the same as the content of local file "CUSTOM/lorem.txt"
-    And as "Alice" the content of "CUSTOM/new-lorem.txt" in the server should be the same as the content of local file "CUSTOM/new-lorem.txt"
+    And as "Alice" the content of "CUSTOM/sub1/lorem.txt" in the server should be the same as the content of local file "CUSTOM/sub1/lorem.txt"
+    And as "Alice" the content of "CUSTOM/sub1/new-lorem.txt" in the server should be the same as the content of local file "CUSTOM/sub1/new-lorem.txt"
+    And as "Alice" the content of "CUSTOM/sub2/sub3/new-lorem.txt" in the server should be the same as the content of local file "CUSTOM/sub2/sub3/new-lorem.txt"
+
+  Scenario: simple upload of a folder that does not exist before with empty sub-folders
+    Given a folder "CUSTOM" has been created with the following files in separate sub-folders in the middleware
+      | subFolder      | file |
+      | sub4           |      |
+      | sub5/sub6/sub7 |      |
+    When the user uploads folder "CUSTOM" using the webUI
+    Then no message should be displayed on the webUI
+    And folder "CUSTOM" should be listed on the webUI
+    And as "Alice" folder "CUSTOM" should exist in the server
+    And as "Alice" file "CUSTOM/sub4" should not exist in the server
+    And as "Alice" file "CUSTOM/sub5/sub6/sub7" should not exist in the server
```

Comment:
```suggestion
    And as "Alice" folder "CUSTOM/sub4" should exist in the server
    And as "Alice" folder "CUSTOM/sub5/sub6/sub7" should exist in the server
```

IMO we should write the test the way the UI is expected to work, so that a developer can get it to pass, and then remove the line from expected-failures.

Reply:
fixed